src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPoolManager.cs

```diff
@@ -299,6 +319,26 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, bool doRe
             return SendAsyncCore(request, proxyUri, doRequestAuth, isProxyConnect:false, cancellationToken);
         }
 
+        public async Task<HttpResponseMessage> SendAsyncMultiProxy(HttpRequestMessage request, bool doRequestAuth, MultiProxy multiProxy, CancellationToken cancellationToken)
+        {
+            HttpRequestException rethrowException = null;
+
+            foreach (Uri proxyUri in multiProxy)
+            {
+                try
+                {
+                    return await SendAsyncCore(request, proxyUri, doRequestAuth, isProxyConnect: false, cancellationToken).ConfigureAwait(false);
+                }
+                catch (HttpRequestException ex) when (ex.AllowRetry != RequestRetryType.NoRetry)
+                {
+                    rethrowException = ex;
+                }
+            }
+            
+            ExceptionDispatchInfo.Throw(rethrowException);
+            return null; // should never be reached.
```

Comment:
Is the compiler forcing you to add the 'return null' ?  

If the code shouldn't be reach, it would be useful to add a Debug.Fail() etc.

Reply:
It is.

Reply:
The compiler doesn't know that EDI.Throw always throws.